activerecord/lib/active_record/persistence.rb

```diff
@@ -380,6 +380,11 @@ def destroy!
     # Any change to the attributes on either instance will affect both instances.
     # If you want to change the sti column as well, use #becomes! instead.
     def becomes(klass)
+      unless self.class <= klass || klass < self.class || self.class.table_name == klass.table_name
```

Comment:
`self.class <=> klass || ....` is better than this. I'll change this line after getting some feedback.

Reply:
There is a `descendants` method you might be able to take advantage of here that goes from the root and lists all the children.